api/src/pcapi/core/api_token/api_token.py

```diff
@@ -0,0 +1,69 @@
+from datetime import datetime
+from datetime import timedelta
+import enum
+
+from flask import current_app as app
+
+from pcapi.core.users.exceptions import InvalidToken
+from pcapi.core.users.utils import decode_jwt_token
+from pcapi.core.users.utils import encode_jwt_payload
+
+
+class TokenType(enum.Enum):
+    TEST_TOKEN_TYPE_1 = "test1"  # this type is used in the tests only
+    TEST_TOKEN_TYPE_2 = "test2"  # this type is used in the tests only
+
+
+class ApiToken:
+    token_type: TokenType
+    user_id: str
+    encoded_token: str
+    data: dict
+
+    def __init__(self, encoded_token: str):
+        self.encoded_token = encoded_token
+        try:
+            payload = decode_jwt_token(encoded_token)
+            self.data = payload["data"] if payload["data"] is not None else {}
+            self.token_type = TokenType(payload["token_type"])
+            self.user_id = payload["user_id"]
+        except Exception:
+            raise InvalidToken()
+
+    def get_data(self) -> dict:
+        return self.data
+
+    def get_user_id(self) -> str:
+        return self.user_id
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**🎯 suggestion:** ‏Les user id sont des int, on peut garder ça ici